sktime/transformers/summarise.py

```diff
@@ -215,3 +228,286 @@ def transform(self, X, y=None):
+                self.passed_default_fc_params = tsfresh.feature_extraction.settings.ComprehensiveFCParameters()
+            elif self.default_fc_parameters == MINIMAL:
+                self.passed_default_fc_params = tsfresh.feature_extraction.settings.MinimalFCParameters()
+            elif self.default_fc_parameters == EFFICIENT:
+                self.passed_default_fc_params = tsfresh.feature_extraction.settings.EfficientFCParameters()
+            else:
+                raise ValueError("Unsupported default_fc_parameters, choose one of COMPREHENSIVE,MINIMAL or EFFICIENT")
+
+
+        elif isinstance(self.default_fc_parameters,
+            (tsfresh.feature_extraction.settings.ComprehensiveFCParameters,
+            tsfresh.feature_extraction.settings.MinimalFCParameters,
+            tsfresh.feature_extraction.settings.EfficientFCParameters)):
+            self.passed_default_fc_params = self.default_fc_parameters
+
+            # TODO checks to be performed over custom parameters
+        elif isinstance(self.default_fc_parameters,dict):
+            if len(self.default_fc_parameters) == 0:
+                self.passed_default_fc_params = None
+            else:
+                self.passed_default_fc_params = self.default_fc_parameters
+
+        else:
+            raise ValueError("Invalid type of default_fc_parameters")
+
+    def fit(self,X, y=None):
+        #empty
+        return self
+
+    def transform(self, X, y=None):
+        """Transform X.
+        Parameters
+        ----------
+        X : univariate nested pandas DataFrame of shape [n_samples, n_columns]
+            Nested dataframe with time-series in cells.
+        Returns
+        -------
+        Xt : pandas DataFrame
+          Transformed pandas DataFrame
+        """
+        #input checks
+        validate_X(X)
+        # check_X_is_univariate(X)
+        X_time_series = convert_data(X)
+       
+
+        # check for default_fc_parameters
+        # TODO Is this method required? Drop handling this error to tsfresh?
+        self._check_default_rc_parameters()
+        
+        # TODO Checks for kind_to_fc_params
+
+        Xt = extract_features(
+                    X_time_series,
+                    column_id="index", column_value="value", 
+                    column_kind="column", column_sort="time_index",
+                    default_fc_parameters=self.passed_default_fc_params,
+                    kind_to_fc_parameters=self.passed_kind_to_fc_params,
+                    n_jobs=self.n_jobs, show_warnings=self.show_warnings,
+                    disable_progressbar=self.disable_progressbar,
+                    impute_function=self.impute_function,
+                    profile=self.profile,
+                    profiling_filename=self.profiling_filename,
+                    profiling_sorting=self.profiling_sorting,
+                    distributor=self.distributor)
+
+        return Xt
+
+
+    def transform_with_input(self,X,y=None):
```

Comment:
I don't think this should be a separate method. If you want to include this, I would add a boolean kwarg in the constructor so that the transformer either return the extracted features or the extracted features together with the original data.